### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by matviichuk92**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,27 @@ public Integer getLength() {
...
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null || object.getClass() != this.getClass()) {
+            return false;
+        }
+        Rectangle figure = (Rectangle) object;
+        return width == null && length == null
+                && color == null ? figure.color == null : color.equals(figure.color);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: we should compare all fields

**Comment by matviichuk92** [comment]
done

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
...
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null || object.getClass() != this.getClass()) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: making the same check twice

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,31 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null || object.getClass() != this.getClass()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: do we have to use `this ` keyword?

**Comment by matviichuk92** [comment]
done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**